akka-http-core/src/main/scala/akka/http/impl/model/parser/UriParser.scala

```diff
@@ -127,23 +128,23 @@ private[http] final class UriParser(val input: ParserInput,
 
   def `relative-part` = rule(
     '/' ~ '/' ~ authority ~ `path-abempty`
-    | `path-absolute`
-    | `path-noscheme`
-    | `path-empty`)
+      | `path-absolute`
+      | `path-noscheme`
+      | `path-empty`)
 
   def scheme = rule(
     'h' ~ 't' ~ 't' ~ 'p' ~ (&(':') ~ run(_scheme = "http") | 's' ~ &(':') ~ run(_scheme = "https"))
-    | clearSB() ~ ALPHA ~ appendLowered() ~ zeroOrMore(`scheme-char` ~ appendLowered()) ~ &(':') ~ run(_scheme = sb.toString))
+      | clearSB() ~ ALPHA ~ appendLowered() ~ zeroOrMore(`scheme-char` ~ appendLowered()) ~ &(':') ~ run(_scheme = sb.toString))
 
-  def `scheme-pushed` = rule { oneOrMore(`scheme-char` ~ appendLowered()) ~ run(_scheme = sb.toString) ~ push(_scheme)}
+  def `scheme-pushed` = rule { oneOrMore(`scheme-char` ~ appendLowered()) ~ run(_scheme = sb.toString) ~ push(_scheme) }
 
   def authority = rule { optional(userinfo) ~ hostAndPort }
 
   def userinfo = rule {
-    clearSB() ~ zeroOrMore(`userinfo-char` ~ appendSB()| `pct-encoded`) ~ '@' ~ run(_userinfo = sb.toString)
+    clearSBForDecoding() ~ zeroOrMore(`userinfo-char` ~ appendSB() | `pct-encoded`) ~ '@' ~ run(_userinfo = getDecodedString())
```

Comment:
Oh man, so simple -- I totally missed that, even though that I was staring at getDecodedString quite a bit. Focused too much on it, and not that it wasnt actually called in `userinfo` it seems

Reply:
Yeah, no worries, it's a bit weird how it is done right here next to parsing but I guess it was done so for performance reasons.